yasnippet.el

```diff
@@ -2672,6 +2672,7 @@ and `kill-buffer' instead."
                  (setq continue (and choose (y-or-n-p "Show also non-active tables? ")))))
              (yas--create-snippet-xrefs)
              (help-mode)
+             (setq-local show-trailing-whitespace nil)
```

Comment:
`setq-local` doesn't exit on older emacsen. 

Also I think the right fix is not to avoid inserting trailing whitespace.